csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs

```diff
@@ -626,6 +671,44 @@ private static bool CheckCudaExecutionProviderDLLs()
 
 
         #endregion
+
+        #region IDisposable Support
+        private bool disposed_ = false; // To detect redundant calls
+
+        /// <summary>
+        /// Public implementation of Dispose pattern callable by consumers.
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            // TODO: uncomment the following line if the finalizer is overridden above.
+            // GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Protected implementation of Dispose pattern.
+        /// </summary>
+        /// <param name="disposing">indicates whether the method call comes from a Dispose method (its value is true) or from a finalizer (its value is false)</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposed_)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                // TODO: dispose managed state (managed objects).
+            }
+
+            // free unmanaged resources(unmanaged objects)
+            NativeMethods.OrtReleaseSessionOptions(handle);
+            handle = IntPtr.Zero;
+
+            disposed_ = true;
+        }
```

Comment:
SafeHandle already implements Dispose() and it calls ReleaseHandle below(). Please, remove this code.
Implementing IDisposable() is not for the faint of heart and one is advised to read the documentation first.

Reply:
Removed.